sigstore-java/src/main/java/dev/sigstore/bundle/BundleVerifier.java

```diff
@@ -15,24 +15,125 @@
  */
 package dev.sigstore.bundle;
 
+import com.google.api.FieldBehavior;
+import com.google.api.FieldBehaviorProto;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.MessageOrBuilder;
+import com.google.protobuf.util.JsonFormat;
 import dev.sigstore.proto.bundle.v1.Bundle;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 
-/**
- * Verifies if Sigstore Bundle is valid.
- *
- * @see <a href="https://github.com/sigstore/protobuf-specs">Sigstore Bundle Protobuf
- *     specifications</a>
- */
+/** Implements Sigstore Bundle verification. */
 public class BundleVerifier {
+  static final JsonFormat.Parser JSON_PARSER = JsonFormat.parser();
+
   /**
-   * Verify if all required fields are initialized in the input protobuf message.
+   * Parses Sigstore Bundle JSON into protobuf.
    *
-   * @param bundleJson message to verify
-   * @return list of all the missing fields
+   * @param bundleJson input JSON
+   * @return parsed bundle
+   * @throws IllegalArgumentException if the JSON is invalid
    */
-  public static List<String> findMissingFields(String bundleJson) {
-    Bundle.Builder bundle = BundleVerifierInternal.parseSigstoreBundle(bundleJson);
-    return BundleVerifierInternal.findMissingFields(bundle);
+  public static List<String> allMissingFields(String bundleJson) {
+    var bundle = Bundle.newBuilder();
+    try {
+      JSON_PARSER.merge(bundleJson, bundle);
+    } catch (InvalidProtocolBufferException e) {
+      throw new IllegalArgumentException("Unable to parse Sigstore Bundle " + bundleJson, e);
+    }
+    return findMissingFields(bundle);
+  }
+
+  static List<String> findMissingFields(MessageOrBuilder message) {
+    final List<java.lang.String> missing = new ArrayList<>();
```

Comment:
what's with the FQ String ?

Reply:
no clue what happened here. Maybe something weird happened when copying?